cmake/Dependencies.cmake

```diff
@@ -824,7 +824,8 @@ if(USE_FBGEMM)
     if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13.0.0)
       # See https://github.com/pytorch/pytorch/issues/74352
       target_compile_options(asmjit PRIVATE -Wno-deprecated-copy)
-      if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13.1.0)
+      if(("${CMAKE_CXX_COMPILER_ID}" MATCHES "AppleClang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.1.6)
+        OR ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0.0))
```

Comment:
This check in line 828 (without _EQUAL) is already in the previous level's if in line 824, so it's guaranteed to be true.

Reply:
yes, I leave it in for clarity (to make clear the conditional is branching on AppleClang vs. Clang)

Reply:
And "AppleClang" will also match "Clang", so `-Wno-unused-but-set-variable` will be added for "Apple clang version 13.0.0 (clang-1300.0.29.30)", resulting in error.

Reply:
lol you are right, I am bad

Reply:
okay it should be fixed now